bkt: init at version 0.6.1

MangoIV opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

- packages https://github.com/dimo414/bkt
- release notes for version 0.6.1 can be found at https://github.com/dimo414/bkt/releases/tag/0.6.1
- adds mangoiv to the maintainers of the package

###### Things done

> **Note**
> I strike out things that do not apply to this PR

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- ~~For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))~~
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- ~~Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)~~
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - ~~(Package updates) Added a release notes entry if the change is major or breaking~~
  - ~~(Module updates) Added a release notes entry if the change is significant~~
  - ~~(Module addition) Added a release notes entry if adding a new NixOS module~~
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

MangoIV commented on pkgs/tools/misc/bkt/default.nix:
```diff
@@ -0,0 +1,30 @@
+  src = fetchFromGitHub {
+    owner = "dimo414";
+    repo = pname;
+    rev = "4b2e3b47e9ca4fdcdc82a01e5ec733a389f81e13";
```
inb4: this is necessary because the version change in the `Cargo.lock` is not included in the release for `0.5.4`, the diff between this commit and `0.5.4` is exactly that update. See https://github.com/dimo414/bkt/commit/4b2e3b47e9ca4fdcdc82a01e5ec733a389f81e13.

dimo414 replied:
0.6.0 is now out which should not have this issue.

MangoIV commented on pkgs/tools/misc/bkt/default.nix:
```diff
@@ -0,0 +1,30 @@
+, fetchFromGitHub
+, lib
+, doCheck ? true
+, withFeatures ? p: p
```
In case there will be a default set of features, this can be used to override or extent this default set; if we have a list we can only replace *or* append.

dali99 replied:
Can't you just use `overrideAttrs` like on all other packages? I'm also not entirely sure how features interact with cargoHash, surely if features pull in more dependencies this hash would also need updating anyways?

MangoIV replied:
> surely if features pull in more dependencies this hash would also need updating anyways

yes, I think this is the case

> Can't you just use overrideAttrs like on all other packages

I'm not sure; the issue is that `overrideAttrs` overrides the set that is passed to `mkDerivation` and not to `buildRustPackage`, I had some issues with this in the past.

MangoIV replied:
Hi, any news here? Should I change this?

dimo414 commented:
Bumping this PR, it'd be awesome to merge this!

MangoIV commented:
@dimo414 refreshed the version.

MangoIV commented:
@dali99, should I request a review from another maintainer? This has been stale for quite some while now.

dimo414 commented:
@MangoIV if you're able to reroute this to another maintainer I think that would be good. Thank you!

MangoIV commented:
I can try and do a request in the nixos server

geri1701 reviewed (changes requested):
`adds mangoiv to maintainers` should be a separate commit. Please rebase and separate commit and force push. You should have two commits here `adds mangoiv to maintainers ` and `bkt: init at version 0.6.1`.

Commit:
adds mangoiv to maintainers

Commit:
bkt: init at version 0.6.1
- packages https://github.com/dimo414/bkt
- release notes for version 0.6.1 can be found at https://github.com/dimo414/bkt/releases/tag/0.6.1

geri1701 commented:
Result of `nixpkgs-review pr 218803` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bkt</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/965

SuperSandro2000 commented on pkgs/tools/misc/bkt/default.nix:
```diff
@@ -0,0 +1,25 @@
+{ rustPlatform
+, fetchFromGitHub
+, lib
+}: rustPlatform.buildRustPackage rec {
```
```suggestion
}:

rustPlatform.buildRustPackage rec {
```

SuperSandro2000 commented on pkgs/tools/misc/bkt/default.nix:
```diff
@@ -0,0 +1,25 @@
+    homepage = "https://github.com/dimo414/bkt";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.mangoiv ];
+    mainProgram = "bkt";
```
This is the default already and does not and should not be repeated

MangoIV commented:
@SuperSandro2000 I'm sorry but this is just ridiculous, why did you not comment in the three months that this was open?

Kranzes commented:
> @SuperSandro2000 I'm sorry but this is just ridiculous, why did you not comment in the three months that this was open?

Sandro doesn't owe you anything and no one is getting paid to review your code. Also there are 4.2K open PRs right now and to think yours should be getting a special treatment is absurd. Just be grateful someone is looking at your code.

MangoIV commented:
I'm not saying that anywhere? I just don't like the behaviour of coming to nitpick after the fact.

MangoIV commented:
> Just be grateful someone is looking at your code.

I am and I thanked you for doing so.

Kranzes commented:
I am talking about you calling Sandro's late review "ridiculous" and "nitpick" instead of thanking them.

MangoIV commented:
Why would I thank them? It's non-constructive, what should I do here? Reopen the PR and remove the arguably unnecessary duplication of a name? I'm not here to get educated or something.

chkno commented:
Today I learned that `mainProgram` need not and should not be specified if it's the same as `pname`.  Thank you, SuperSandro2000!

MangoIV: The `nixfmt` tool automatically formats Nix files, freeing you from ever thinking about formatting concerns.  Editing .nix files haphazardly without thought to formatting and then letting the tool take care of the formatting is a nice experience.  The only minor trouble is that this can introduce spurious whitespace diffs if new files / large changes are not machine-formatted.  Some amount of this churn is fine and normal, but we can try to keep it to a minimum.

--

This PR had two very minor errors.  That's totally fine!  I erred in approving it with these minor errors in it, perhaps.  That's also totally fine.  We both learned a thing.  I think PR threads are a place for getting educated, and that this is a good thing.

MangoIV: You do not need to re-open this PR, or create another PR to address these nitpicks.  Think of them as FYIs, to know for next time, to fix incidentally next time you're editing this file maybe, or to know when reviewing others' PRs.

Every nix package is a working example of how to package a piece of software.  Folks look at / copy from existing package definitions when adding new packages.  In this way, keeping things tidy makes others' future contributions easier & smoother.

MangoIV, SuperSandro2000, Kranzes, dali99, dimo414, geri1701: Thank you all for all of your contributions!
